src/main/java/cloud/orbit/spring/test/OrbitSpringTestConfigurationChecker.java

```diff
@@ -0,0 +1,57 @@
+/*
+ Copyright (C) 2017 Electronic Arts Inc.  All rights reserved.
+
+ Redistribution and use in source and binary forms, with or without
+ modification, are permitted provided that the following conditions
+ are met:
+
+ 1.  Redistributions of source code must retain the above copyright
+     notice, this list of conditions and the following disclaimer.
+ 2.  Redistributions in binary form must reproduce the above copyright
+     notice, this list of conditions and the following disclaimer in the
+     documentation and/or other materials provided with the distribution.
+ 3.  Neither the name of Electronic Arts, Inc. ("EA") nor the names of
+     its contributors may be used to endorse or promote products derived
+     from this software without specific prior written permission.
+
+ THIS SOFTWARE IS PROVIDED BY ELECTRONIC ARTS AND ITS CONTRIBUTORS "AS IS" AND ANY
+ EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ DISCLAIMED. IN NO EVENT SHALL ELECTRONIC ARTS OR ITS CONTRIBUTORS BE LIABLE FOR ANY
+ DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package cloud.orbit.spring.test;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Configuration;
+
+import cloud.orbit.actors.cluster.ClusterPeer;
+import cloud.orbit.spring.OrbitSpringConfiguration;
+
+@Configuration
+@ConditionalOnProperty("org.springframework.boot.test.context.SpringBootTestContextBootstrapper")
+@ConditionalOnMissingBean(ClusterPeer.class)
+@AutoConfigureAfter(OrbitSpringTestConfiguration.class)
+public class OrbitSpringTestConfigurationChecker implements InitializingBean
```

Comment:
Not sure the purpose of this class. If the `OrbitSpringTestConfiguration` makes a ClusterPeer then what use cases would you see this code being run?

Reply:
This class is here solely for the purpose of logging a warning if all 3 of these conditions are met:
1. User is running a SpringBootTest
2. User has included orbit-spring configuration in the test
3. User does _not_ have FakeClusterPeer in their classpath.

The reason a user would not have FakeClusterPeer in their classpath is if they did not include orbit-actor-tests as a test dependency. Unfortunately, FakeClusterPeer resides in the orbit-actor-tests library, and it doesn't make sense to include a test library as a default-scope dependency of orbit-spring.

Reply:
Notice that OrbitSpringTestConfiguration is `@ConditionalOnClass(FakeClusterPeer.class)`